nodejs/week2/nodejsweekwork/src/backend/index.js

```diff
@@ -0,0 +1,41 @@
+const express = require("express");
+const router = express.Router();
+const app = express();
+
+const meals = require("./routes/meals");
+const reservations = require("./routes/reservations");
+const reviews = require("./routes/reviews.js");
+const cheapMeals = require("./routes/cheap-meals.js");
+const largeMeals = require("./routes/large-meals.js");
+const randommeal = require("./routes/randommeal.js");
+const randomReservation = require("./routes/randomReservation");
+
+const mealswithID = require("./routes/mealswithID.js");
+const reservationsWithID = require("./routes/reservationsWithID");
+const reviewsWithID = require("./routes/reviewsWithID.js");
+const mealPrice = require("./routes/mealPrice");
+const mealsTitle = require("./routes/mealsTitle");
+const mealsCreatedAfter = require("./routes/mealsCreatedAfter");
+const mealsLimit = require("./routes/mealsLimit");
+
+app.use("/meals", meals);
+app.use("/reservations", reservations);
+app.use("/reviews", reviews);
+app.use("/cheap-meals", cheapMeals);
+app.use("/large-meals", largeMeals);
+app.use("/randommeal", randommeal);
+app.use("/randomReservation", randomReservation);
+
+app.use("/meals", mealswithID);
+app.use("/reservationsid", reservationsWithID);
+app.use("/reviewsid", reviewsWithID);
+app.use("/meals/mealPrice", mealPrice);
+app.use("/meals/title", mealsTitle);
+app.use("/meals/createdAfter", mealsCreatedAfter);
+app.use("/meals/limit", mealsLimit);
```

Comment:
I think that you may be mixing a couple of concepts here. The way that this is implemented right now, it would make more sense to either:

1) NOT use routers and only rely on app.get

```
// in index.js
app.get('/meals', mealsRouter)     // => mealsRouter = (req, res) => {.....}
app.get('/meals/:id', mealRouter) // likewise
// in mealsRouter.js
module.exports = (req, res) => {
  // return some meals
};
```

where the `mealRouter` is just a request handler that can be imported from a separate module. This is a simple approach that doesn't require the user of middleware. 

2) Use routers to sort of break down parts of the url with that can be handled by a specific router:

```
// in index.js
app.use('/meals', mealsRouter);
// in mealsRouter.js
const router = express.Router();
router.get('/', (req, res) => { /* return meals */ }
router.get('/:id', (req, res) => { /* return a meal by id */ }
```
This can be useful when your application starts growing because you sort of break down the application into smaller parts that handle different routes and do different things. It is slightly more complex because you add one new component into the mix: the router.

For now, I would encourage you to use `app.get` because of its simplicity, and later refactor to using routers when you feel comfortable with `app.get`.